app/services/clients/adminusers.client.js

```diff
@@ -51,101 +47,69 @@ module.exports = function (clientOptions = {}) {
    * @return {Promise<User>} A promise of a User
    * @param externalIds
    */
-  function getUsersByExternalIds (externalIds = []) {
-    return baseClient.get(
-      {
-        baseUrl,
-        url: `${userResource}`,
-        qs: {
-          ids: externalIds.join()
-        },
-        json: true,
-        description: 'find a user',
-        service: SERVICE_NAME,
-        transform: responseBodyToUserListTransformer
-      }
-    )
+  async function getUsersByExternalIds (externalIds = []) {
+    const url = `${baseUrl}${userResource}?ids=${externalIds.join()}`
+    this.client = new Client(SERVICE_NAME)
```

Comment:
Using `this` in this context seems to be unhelpful.  I'd suggest using `const client =` instead.

Reply:
Having got further through the code I'm wondering whether you'd prefer to use a shared client.  I'm not sure from just reading the code whether that's a suitable solution or not but it seems like it might be a good match as the client seems to need configuring before use and the configuration seems to be generic.

Reply:
Agreed, this approach seems better. Code changed as suggested.